src/plugins/intel_gpu/src/kernel_selector/cl_kernels/gemm_tiled_opt.cl

```diff
@@ -153,76 +186,65 @@ KERNEL(gemm_tiled_opt)(
         // Loading B tile
         unroll_for (uint b_load_id = 0; b_load_id < TILE_K; b_load_id++) {
 #if IS_DYNAMIC
+#if TRANSPOSE_INPUT1 == 0
 #if HAS_DYNAMIC_N_PADDING
-            // In case of dynamic padding we can't guarantee memory access alignment for
-            // block reads (4 bytes), so use scattered read
             b_tile[b_load_id] = b_raw_global_id > N - 1 ? 0 : b_ptr[sglid];
 #else
             b_tile[b_load_id] = TILE_N_NOT_DIVISIBLE ? (b_raw_global_id > N - 1 ? 0 : b_ptr[sglid]) : BLOCK_READ_B(b_ptr, 0);
 #endif
+            b_ptr += input1_offset;
+#elif TRANSPOSE_INPUT1 == 2 // TRANSPOSE_INPUT1 == 0
+            if (b_raw_global_id > N - 1) {
+                b_tile[b_load_id] = 0;
+            } else {
+                uint b_idx = FUNC_CALL(get_input1_index)(OPTIONAL_SHAPE_INFO_TENSOR b, f, w, z, (b_load_id + k * TILE_K), x);
+                b_tile[b_load_id] = input1[b_idx];
+            }
+#endif // TRANSPOSE_INPUT1 == 0
 #else // IS_DYNAMIC
+#if TRANSPOSE_INPUT1 == 0
 #if TILE_N_NOT_DIVISIBLE
             b_tile[b_load_id] = b_raw_global_id > N - 1 ? 0 : b_ptr[sglid];
 #else // TILE_N_NOT_DIVISIBLE
             b_tile[b_load_id] = BLOCK_READ_B(b_ptr, 0);
 #endif // TILE_N_NOT_DIVISIBLE
+            b_ptr += input1_offset;
+#elif TRANSPOSE_INPUT1 == 2 // TRANSPOSE_INPUT1 == 0
```

Comment:
0/1/2 is very not intuitive. Could you make a better macro?

Reply:
I added three macros, `TRANSPOSE_X_LAST` , `TRANSPOSE_Y_LAST`  and `TRANSPOSE_OTHER` for better readability. Thank you!